### ISSUE TITLE
[BUG] Network variable exception

### ISSUE THREAD

**Issue by Lennord** [Zulfen/ZulfBungee#7]
**Describe the bug** and **To reproduce**
1. Installation of the bungeecord server and the spigot server on a VPS.
2. Installation of the Skript plugin and your Addon.
3. Then configure the 2 configs and I turn on the servers then it gives me the error and it spams me.

**Expected behavior**
Following the configuration, normally it should connect.

**Screenshots**
![Capture](https://user-images.githubusercontent.com/65398078/216100140-53e522f0-6387-4f55-b8ca-b87d5b0f5b68.JPG)

**Spigot server information (in this order)**
 - Plugin version Alpha 0.7.9
 - Minecraft Version: Paper-1.18.2-388
 - Skript Version 2.6.4
 - Java Version 17

**Proxy information (in this order)**
 - Proxy Type Waterfall-1.19-510
 - Plugin Version Alpha 0.7.9
 - Minecraft Version: 1.18.2
 - Java Version 17

**Additional context**
Plugin Bungeecord File config.yml: https://pastebin.com/uxy8Wt3f
Plugin Spigot File config.yml: https://pastebin.com/G1MrmA4F

**Comment by Zulfen** [comment]
Hi! Interesting bug, haven't seen this one yet so thanks for reporting :)
Would you mind turning debug mode on to see what happens both on the Bungeecord and Spigot and send the logs if that's alright?

**Comment by Zulfen** [comment]
I think it might be related to port whitelisting - try turning that off and see what happens?

**Comment by Lennord** [comment]
> Hi! Interesting bug, haven't seen this one yet so thanks for reporting :) Would you mind turning debug mode on to see what happens both on the Bungeecord and Spigot and send the logs if that's alright?

Bungeecord logs: [latest.log](https://github.com/Zulfen/ZulfBungee/files/10606684/latest.log)
Spigot logs: [latest.log](https://github.com/Zulfen/ZulfBungee/files/10606686/latest.log)

Here are the Bungeecord and Spigot server logs with debug-mode enabled.

**Comment by Lennord** [comment]
> I think it might be related to port whitelisting - try turning that off and see what happens?

I still have the same problem with and without whitelist-ports.

**Comment by Zulfen** [comment]
Hm.. okay!
Have you put your spigot server in Bungeecord's config.yml? Not the plugin, but the actual Bungeecord config file.
Could you also share that as well? Thanks!

**Comment by Lennord** [comment]
> Hm.. okay! Have you put your spigot server in Bungeecord's config.yml? Not the plugin, but the actual Bungeecord config file. Could you also share that as well? Thanks!

Yes, the Hub01 server has been put on the bungeecord configuration.
Config.yml: https://pastebin.com/X7uzujPU

**Comment by Zulfen** [comment]
May have found the issue - try setting `192.168.1.18` as the address instead of `localhost` in the bungeecord config.
If this works - I will fix the plugin so that it recognises localhost as well as the numerical IP :]

**Comment by Zulfen** [comment]
In the newest version, I managed to reproduce the bug you got and it should be fixed now!

**Comment by Lennord** [comment]
> In the newest version, I managed to reproduce the bug you got and it should be fixed now!

Thank you, the problem is solved with the "Alpha 0.8.0".
On the other hand, I don't know if it's a bug but when I turn on the bungeecord and the spigot server it works but when I restart or I was and I turn on the spigot server and the bungeecord is already on, it sends me this message in red and if I wait 1 or 2 minutes it works (it reconnects).
Is there a way for it to reconnect directly or not?
![Capture](https://user-images.githubusercontent.com/65398078/216810416-a4647a81-19e6-46a8-850c-8f694feadbb8.JPG)

**Comment by Lennord** [comment]
I specify that there is no error on the bungeecord and that I activated the debug-mode on the bungeecord and the spigot server.

**Comment by Lennord** [comment]
> In the newest version, I managed to reproduce the bug you got and it should be fixed now!

The problem is solved, thank you very much it helps me enormously and to those who use your plugin.

**Comment by Lennord** [comment]
After this problem solved, I have a problem with the network variables only when you do "on join" or "on connect".
I made the code below and when I join the server it shows me the error in the server console below.

Bug Report latest logs: https://pastebin.com/zp5U5TA9
Codes: 
![Code](https://user-images.githubusercontent.com/65398078/216812006-71542306-ad32-43ba-8e03-933184409d87.JPG)

**Comment by Zulfen** [comment]
Hi! Thanks for the report - does this consistently happen or does this only happen sometimes?

**Comment by Lennord** [comment]
> Hi! Thanks for the report - does this consistently happen or does this only happen sometimes?

it happens sometimes

**Comment by Zulfen** [comment]
Hm.. will look into this!

**Comment by Lennord** [comment]
> 

Ok thank you.

> Hm.. will look into this!

Ok thank you.

**Comment by Lennord** [comment]
Hello, I would like to know if the network variable will be corrected in a few days because I really need it but don't worry I'll wait, I just wanted news of the bug.

**Comment by Zulfen** [comment]
This should be fixed now I hope :)

**Comment by Lennord** [comment]
> This should be fixed now I hope :)

The problem is fixed with 0.8.1, I tested everything and it works perfectly, thank you very much :)

**Comment by Zulfen** [comment]
No worries! If the issue happens again, let me know :]

**Comment by Lennord** [comment]
After this problem, I found another problem again on the network variable but a small bug.

when I set a network variable with a number and I add or remove a number in this variable, this gives me several number characters but which does not give the correct number.
there is no error either on the bungee or on the spigot with or without debug mode.

Also when I do in a function which works perfectly:
`set network variable {account::%player's uuid%::rankauthor} to {_s}`
it gives me this error: https://pastebin.com/CtDYqAmL
